manager/watcher/rkt/rkt.go

```diff
@@ -0,0 +1,154 @@
+// Copyright 2014 Google Inc. All Rights Reserved.
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//     http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+// Package container defines types for sub-container events and also
+// defines an interface for container operation handlers.
+package rkt
+
+import (
+	"path/filepath"
+	"time"
+
+	"github.com/google/cadvisor/container/rkt"
+	"github.com/google/cadvisor/manager/watcher"
+
+	"github.com/golang/glog"
+	"golang.org/x/net/context"
+
+	rktapi "github.com/coreos/rkt/api/v1alpha"
+)
+
+type rktContainerWatcher struct {
+	// Signal for watcher thread to stop.
+	stopWatcher chan error
+}
+
+func NewRawContainerWatcher() (watcher.ContainerWatcher, error) {
+	watcher := &rktContainerWatcher{
+		stopWatcher: make(chan error),
+	}
+
+	return watcher, nil
+}
+
+func (self *rktContainerWatcher) Start(events chan watcher.ContainerEvent) error {
+	go self.detectRktContainers(events)
+
+	return nil
+}
+
+func (self *rktContainerWatcher) Stop() error {
+	// Rendezvous with the watcher thread.
+	self.stopWatcher <- nil
+	return <-self.stopWatcher
+}
+
+func (self *rktContainerWatcher) detectRktContainers(events chan watcher.ContainerEvent) {
+	glog.Infof("starting detectRktContainers thread")
+	ticker := time.Tick(30 * time.Second)
+	curpods := make(map[string]*rktapi.Pod)
+
+	for {
+		select {
+		case _ = <-ticker:
+			glog.Infof("rkt ticker")
+			pods, err := ListRunningPods()
+			if err != nil {
```

Comment:
I'm not sure if this is just always running, but in the relatively common case of the rkt api server not being reachable, I don't think we should spam the log every 30s. Will the rkt code only be run with user intent? Should there be code before that tries to determine if the rkt api server is up and errors out once early? Is this concern invalid for whatever reason?